qualang_tools/addons/calibration/README.md

```diff
@@ -0,0 +1,167 @@
+4. One needs to open a quantum machine manager with the correct IP address.
+5. The calibration can be simulated to check the validity of the sequences. This will plot the analog and digital pulses played during the sequences.
+6. After opening a quantum machine with the configuration, the calibrations can be executed. The flag called `plot` can be take three values:
+   * "full" to plot the results at the end of the sequence.
+   * "live" to plot the results while the sequence is being executed.
+   * None which will not plot anything.
+7. The results are stored in a dictionary accessible under `my_calib.results`.
+
+
+```python
+from qm.QuantumMachinesManager import QuantumMachinesManager
+from qualang_tools.addons.calibration.calibrations_API import *
+from configuration import * 
+
+# Relevant configuration parameters
+#############################################################
+resonator_element = "resonator"
+resonator_operation = "readout"
+qubit_element = "qubit"
+qubit_operation = "cw"
+flux_line_element = None
+flux_line_operation = None
+int_w = ["cos", "sin", "minus_sin", "cos"]
+outs = ["out1", "out2"]
+# Options for plotting figures
+options = {
+    "fontsize": 14,
+    "color": "b",
+    "marker": ".",
+    "linewidth": 1,
+    "figsize": (12, 15),
+}
+#############################################################
+
+# Initialize the calibration API with the relevant configuration parameters
+my_calib = QUA_calibrations(
+    configuration=config,
+    readout=(resonator_element, resonator_operation),
+    qubit=(qubit_element, qubit_operation),
+    integration_weights=int_w,
+    outputs=outs,
+)
+
+# Set calibrations from the ones detailed below
+my_calib.set_...
+
+# Open quantum machines manager
+qmm = QuantumMachinesManager(host="127.0.0.1", port=80)
+
+# Simulate the calibrations
+my_calib.simulate_calibrations(qmm, simulation_duration=5000)
+
+# Run the calibrations
+qm = qmm.open_qm(config)
+my_calib.run_calibrations(quantum_machine=qm, plot="full", plot_options=options)
+```
+
+## Available calibrations
+### Raw traces
+This program acquires the raw and averaged traces from the two inputs of the OPX.
+It only requires two inputs:
+* `iterations`: number of averaging iterations. Must be a positive Python integer.
+* `cooldown_time`: wait time between two iterations in clock cycle unit (4ns). Must be an integer.
+```python
+my_calib.set_raw_traces(iterations=1000, cooldown_time=0)
+```
+![raw_traces](raw_traces.PNG)
+
+### Time of flight
+This program measures the time of flight and offsets of the two inputs of the OPX.
+It requires three inputs:
+* `iterations`: number of averaging iterations. Must be a positive Python integer.
+* `cooldown_time`: wait time between two iterations in clock cycle unit (4ns). Must be an integer.
+* `threshold`: detection threshold for time of flight estimation. The default value is 10/4096 which is usually good enough to detect the waveform.
```

Comment:
In what units is this? =

Reply:
it should be something like V/ns, I ass it